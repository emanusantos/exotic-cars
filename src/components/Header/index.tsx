import React from 'react';
import * as S from './styles';
import { ReactComponent as Calendar } from 'assets/calendar.svg';
import { ReactComponent as Location } from 'assets/location.svg';
import { ReactComponent as Search } from 'assets/search.svg';

const Header = () => {
    return (
        <S.Header>
            <S.Wrapper>
                <S.Title>EXOTIC</S.Title>
                <S.Subtitle>CARS</S.Subtitle>
            </S.Wrapper>
            <S.Wrapper colored>
                <S.Wrapper marginRight="4rem">
                    <Location />
                    <S.Text>North Carolina, NC 90025</S.Text>
                </S.Wrapper>
                <S.Wrapper marginRight="1rem">
                    <Calendar />
                    <S.Text>11/03/2021</S.Text>
                </S.Wrapper>
                <S.Wrapper marginRight="1rem">
                    <Calendar />
                    <S.Text>12/12/2021</S.Text>
                </S.Wrapper>
                <div
                    style={{
                        backgroundColor: '#fff',
                        padding: '.2rem .5rem',
                        borderRadius: '5rem'
                    }}
                >
                    <Search />
                </div>
            </S.Wrapper>
            <S.Wrapper>
                <S.Button>Sign up</S.Button>
                <S.Button primary>Sign in</S.Button>
            </S.Wrapper>
        </S.Header>
    );
};

export default Header;
