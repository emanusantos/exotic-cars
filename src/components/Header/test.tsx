import React from 'react';
import Header from '.';
import { render, screen } from '@testing-library/react';

describe('<Header />', () => {
    it('should render heading', () => {
        const { container } = render(<Header />);
        expect(
            screen.getByRole('heading', { name: /exotic/i })
        ).toBeInTheDocument();

        expect(container.firstChild).toMatchSnapshot();
    });
});
