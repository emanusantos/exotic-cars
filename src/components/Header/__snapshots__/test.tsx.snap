// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should render heading 1`] = `
<header
  class="sc-bdfBQB eXRFvu"
>
  <div
    class="sc-gsTEea bLZMfK"
  >
    <h2
      class="sc-dlfnuX gldEDn"
    >
      EXOTIC
    </h2>
    <h4
      class="sc-hKgJUU CRIFQ"
    >
      CARS
    </h4>
  </div>
  <div
    class="sc-gsTEea hlSGxa"
  >
    <div
      class="sc-gsTEea ehHOvI"
    >
      <div />
      <p
        class="sc-jSgvzq hqPMxm"
      >
        North Carolina, NC 90025
      </p>
    </div>
    <div
      class="sc-gsTEea ehIYQB"
    >
      <div />
      <p
        class="sc-jSgvzq hqPMxm"
      >
        11/03/2021
      </p>
    </div>
    <div
      class="sc-gsTEea ehIYQB"
    >
      <div />
      <p
        class="sc-jSgvzq hqPMxm"
      >
        12/12/2021
      </p>
    </div>
    <div
      style="background-color: rgb(255, 255, 255); padding: .2rem .5rem; border-radius: 5rem;"
    >
      <div />
    </div>
  </div>
  <div
    class="sc-gsTEea bLZMfK"
  >
    <button
      class="sc-eCstlR frUVS"
    >
      Sign up
    </button>
    <button
      class="sc-eCstlR RqSUo"
    >
      Sign in
    </button>
  </div>
</header>
`;
